components/scream/src/physics/spa/tests/spa_main_test.cpp

```diff
@@ -0,0 +1,381 @@
+        REQUIRE(aer_tau_lw_h(dof_i,n,kpack)[kidx] == aer_tau_lw_end(dof_i,n,kpack)[kidx] );
+      }
+    }
+  }
+
+  // Add a few days and update spa data.  Make sure that the output values are not outside of the bounds
+  // of the actual SPA data 
+  ts += (int)(util::days_in_month(ts.get_year(),ts.get_month())*0.5)*86400;
+  SPAFunc::update_spa_timestate(spa_data_file,nswbands,nlwbands,ts,spa_horiz_interp,spa_time_state,spa_data_beg,spa_data_end);
+  Kokkos::deep_copy(ps_beg        ,spa_data_beg.PS);
+  Kokkos::deep_copy(ccn3_beg      ,spa_data_beg.CCN3);
+  Kokkos::deep_copy(aer_g_sw_beg  ,spa_data_beg.AER_G_SW);
+  Kokkos::deep_copy(aer_ssa_sw_beg,spa_data_beg.AER_SSA_SW);
+  Kokkos::deep_copy(aer_tau_sw_beg,spa_data_beg.AER_TAU_SW);
+  Kokkos::deep_copy(aer_tau_lw_beg,spa_data_beg.AER_TAU_LW);
+  Kokkos::deep_copy(ps_end        ,spa_data_end.PS);
+  Kokkos::deep_copy(ccn3_end      ,spa_data_end.CCN3);
+  Kokkos::deep_copy(aer_g_sw_end  ,spa_data_end.AER_G_SW);
+  Kokkos::deep_copy(aer_ssa_sw_end,spa_data_end.AER_SSA_SW);
+  Kokkos::deep_copy(aer_tau_sw_end,spa_data_end.AER_TAU_SW);
+  Kokkos::deep_copy(aer_tau_lw_end,spa_data_end.AER_TAU_LW);
+  // Create a target pressure profile to interpolate onto that has a slightly higher surface pressure that the bounds.
+  // This will force extrapolation.
+  for (size_t dof_i=0;dof_i<dofs_gids_h.size();dof_i++) {
+    for (int kk=0;kk<nlevs;kk++) {
+      int kpack = kk / Spack::n;
+      int kidx  = kk % Spack::n;
+      Real ps = 1.05*std::max(ps_beg(dof_i),ps_end(dof_i));
+      pmid_h(dof_i,kpack)[kidx] = ps*hybm_h(kpack)[kidx] + P0*hyam_h(kpack)[kidx];
+    }
+  }
+  Kokkos::deep_copy(pressure_state.pmid,pmid_h);
+
+  SPAFunc::spa_main(spa_time_state,spa_pressure_state,spa_data_beg,spa_data_end,spa_data_out,dofs_gids.size(),nlevs,nswbands,nlwbands);
+  Kokkos::deep_copy(ccn3_h      , spa_data_out.CCN3);
+  Kokkos::deep_copy(aer_g_sw_h  , spa_data_out.AER_G_SW);
+  Kokkos::deep_copy(aer_ssa_sw_h, spa_data_out.AER_SSA_SW);
+  Kokkos::deep_copy(aer_tau_sw_h, spa_data_out.AER_TAU_SW);
+  Kokkos::deep_copy(aer_tau_lw_h, spa_data_out.AER_TAU_LW);
+
+  // Calculate the min and max values for all spa input data for all columns 
+  auto ccn3_bnds       = view_2d<Real>("",2,ncols);
+  auto aer_sw_g_bnds   = view_3d<Real>("",2,ncols,nswbands);
+  auto aer_sw_ssa_bnds = view_3d<Real>("",2,ncols,nswbands);
+  auto aer_sw_tau_bnds = view_3d<Real>("",2,ncols,nswbands);
+  auto aer_lw_tau_bnds = view_3d<Real>("",2,ncols,nlwbands);
+  auto ccn3_bnds_h       = Kokkos::create_mirror_view(ccn3_bnds      );
+  auto aer_sw_g_bnds_h   = Kokkos::create_mirror_view(aer_sw_g_bnds  );
+  auto aer_sw_ssa_bnds_h = Kokkos::create_mirror_view(aer_sw_ssa_bnds);
+  auto aer_sw_tau_bnds_h = Kokkos::create_mirror_view(aer_sw_tau_bnds);
+  auto aer_lw_tau_bnds_h = Kokkos::create_mirror_view(aer_lw_tau_bnds);
+  for (size_t dof_i=0;dof_i<dofs_gids_h.size();dof_i++) {
```

Comment:
It appears to me that you are doing both time and vertical interpolation, and checking that minmax(X_out) is within minmax of both X_beg and X_end.

I think that's a weak form test. I think you should test time and vertical interpolation separately. For time, you need a pointwise min(X_beg,X_end) < X_out < max(X_beg,X_end), which is a stronger condition than minmax over column begin bounded by minmax of beg/end. 

For vertical interpolation, instead, I would pick X_beg=X_end (for simplicity). I would also consider writing a helper function that computes min/max of a 1d view:
```
template<typename ViewT>
std::pair<Real,Real> minmax (const ViewT& v_out);
```
and loop over idof (and nbands), calling minmax on subviews of src and tgt fields, and checking their values. This might remove some code duplication, and make the intent of the test (min(in)<output<max(in) over column) more obvious to the reader.